penman/__main__.py

```diff
@@ -222,12 +222,16 @@ def main():
         '--indicate-branches', action='store_true',
         help='insert triples to indicate tree structure')
 
+    norm.add_argument(
+        '--encoding', action='store_true', default=locale.getpreferredencoding(),
+        help='encoding to use for input/output. Defaults to system default')
+
     args = parser.parse_args()
 
     if args.quiet:
         args.verbosity = 0
         sys.stdout.close()
-        sys.stdout = open(os.devnull, 'w')
+        sys.stdout = open(os.devnull, 'w', encoding=args.encoding)
```

Comment:
Not sure this one is required, since it's writing to `os.devnull`?

Reply:
I must admit, I am not sure. I don't know if anything is actually written (and encoded) and then discarded, or just never encoded when targeting devnull. I have removed the encoding argument for now.

Reply:
Hmm, the encoding does seem to make a difference:

```pycon
>>> import os
>>> with open(os.devnull, 'w', encoding='cp1252') as devnull:
...     print('符号', file=devnull)
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/usr/lib/python3.10/encodings/cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode characters in position 0-1: character maps to <undefined>
```

Might be best to use a unicode encoding like `'uft-8'`, unless the `open()` function automatically chooses something appropriate for special files like `os.devnull`. I don't see any articles mentioning a best practice here.

Reply:
I believe that there are _some_ [potential exceptions](https://stackoverflow.com/a/47820998/1150683) though since some exceptional sequences cannot be rightly encoded from cp1252 to utf8. But I'd think the most generous way to solve _that_ would be by also adding an optional `output_encoding` argument for this devnull call that users can use in the exceptional case that they run into problems.

I am not a huge fan of that solution (it adds yet another argument) but it does seem the most all-covering solution from a user's perspective. For now I've set it to utf-8 but let me know if you do want to add another CLI argument like `output_encoding` and I'll add that.